### ISSUE TITLE
[Help] Search is working in development; not working in Testflight & AppleStore

### ISSUE THREAD

**Issue by Abacaxi-Nelson** [GetStream/stream-chat-flutter#673]
### Which packages are you using?

stream_chat, stream_chat_persistance

### On what platforms did you experience the issue?

iOS

### What version are you using?

stream_chat_flutter: ^2.2.1
stream_chat_persistence: ^2.2.0

### What happened?

Its weird, but after half day trying to figure it out; i convince my self to ask for some help.
Same app:
![IMG-2365](https://user-images.githubusercontent.com/556990/132860300-0291de0e-751a-4f8a-afb6-b9066e0d4ce8.PNG)
![IMG-2364](https://user-images.githubusercontent.com/556990/132860278-edc56389-32c5-473c-8ca5-2ebd71561d30.jpg)
 
First one is Testflight version, when search a user dont work.
Second screen in my dev env, when search works...

One thing that coud explain, is the api call is not triggered; cause i dont see the log in 'Chat Logs/Query Users' for testflight version; but see it for dev app..

### Steps to reproduce

```bash
Using your example app..
Rest of the app workin great :)
If you have an idea, please share :)
```

### Supporting info to reproduce

```dart
await chatClient.disconnectUser();
      await chatClient.connectUser(
        chat.User(
          id: uid,
          extraData: {
            'image': user.imageUrl ??
                'XXXX',
            'name': user.userName,
          },
        ),
        response.body,
      );
```
Using your example app..

### Flutter doctor output

```shell
Flutter 2.5
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Abacaxi-Nelson** [comment]
really strange.
launching in debug mode work.
launching in release/profile mode dont work: ```queryUsers is not called when name is typed```
https://github.com/GetStream/stream-chat-flutter/blob/18c6fd842c86767461adbd5adfd71770d34922f1/packages/stream_chat_flutter_core/lib/src/users_bloc.dart#L65

**Comment by xsahil03x** [comment]
@Abacaxi-Nelson check if you have the internet permission available in the profile and release flavor.

**Comment by Abacaxi-Nelson** [comment]
Hey @xsahil03x 
I do have it cause rest of the app is working.
Im able to display my chanels, and even send message in them.

**Comment by Abacaxi-Nelson** [comment]
hey @xsahil03x 
i think i narrow down the problem.
here https://github.com/GetStream/stream-chat-flutter/blob/720c0a80edc51f8998f23b39f710b2312d6c9ee1/packages/stream_chat_flutter_core/lib/src/user_list_core.dart#L210
**toString** method is used.

In my test, toString works great in debug bu not in profile/release (it show always 'Filter' no matter Filter is)

i add this method to Filter (https://github.com/GetStream/stream-chat-flutter/blob/develop/packages/stream_chat/lib/src/core/models/filter.dart)

```dart
@override
  String toString() {
    return jsonEncode(toJson());
  }
```

seems to work...
any idea ?

**Comment by xsahil03x** [comment]
Hey @Abacaxi-Nelson
Thanks for the research, I now understand why this is not working for you 👍🏼 
I will soon create a PR for this fix.

**Comment by imtoori** [comment]
@Abacaxi-Nelson I can confirm this is fixed
the fix has been released in 3.0.0
